### PR TITLE
feat(chat): fix app icon publishing warning message, add 'save all' option to code editor and 'save' to code editor file context menu, fix code editor empty files content (Issues #2576, #2515, #2427)

### PR DESCRIPTION
**Pull request by Alexander-Kezik**
**Description:**

fix app icon publishing warning message
add 'save all' option to code editor and 'save' to code editor file context menu
fix code editor empty files content

Issues:

- Issue #2576
- Issue #2515
- Issue #2427

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Alexander-Kezik** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/537341)

**Comment by Alexander-Kezik** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/537350)

File: apps/chat/src/utils/app/data/storages/api/conversation-api-storage.ts [hunk 1]
```diff
@@ -58,12 +58,12 @@ export class ConversationApiStorage extends ApiEntityStorage<
   }
 }
 
-export const getOrUploadConversation = (
-  payload: { id: string },
+export const getOrUploadConversation = <T>(
```
Comment (IlyaBondar) [inline apps/chat/src/utils/app/data/storages/api/conversation-api-storage.ts]:
```suggestion
export const getOrUploadConversation = <T extends { id: string }>(
```

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2603.nightly-test.deltixhub.io
>E2E tests status: success

**Review by IlyaBondar — DISMISSED** [review]

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2603.nightly-test.deltixhub.io
>E2E tests status: success

**Review by IlyaBondar — APPROVED** [review]

**Pull request merged**